homeassistant/components/husqvarna_automower/diagnostics.py

```diff
@@ -0,0 +1,38 @@
+"""Diagnostics support for Husqvarna Automower."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+
+CONF_REFRESH_TOKEN = "refresh_token"
+POSITIONS = "positions"
+
+TO_REDACT = {
+    CONF_ACCESS_TOKEN,
+    CONF_REFRESH_TOKEN,
+    POSITIONS,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    return async_redact_data(entry.as_dict(), TO_REDACT)
+
+
+async def async_get_device_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry, device: DeviceEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a device entry."""
+    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    mower_id = list(next(iter(device.identifiers)))[1]
```

Comment:
I don't think it safe to assume the first one is the identifier you can / should use? Besides it being tuples, it may have multiple entries?